tests/base/test_tensor_innerproduct.py

```diff
@@ -162,6 +162,166 @@ def test_order6_faces_invert_model(self):
         self.orderTest()
 
 
+class TestInnerProductsFaceProperties3D(discretize.tests.OrderTest):
+    """Integrate a function over a surface within a unit cube domain
+    using edgeInnerProducts and faceInnerProducts."""
+
+    meshTypes = ["uniformTensorMesh"]
+    meshDimension = 3
+    meshSizes = [16, 32]
+
+    def getError(self):
+        call = lambda fun, xyz: fun(xyz[:, 0], xyz[:, 1], xyz[:, 2])
+
+        ex = lambda x, y, z: x**2 + y * z
+        ey = lambda x, y, z: (z**2) * x + y * z
+        ez = lambda x, y, z: y**2 + x * z
+
+        tau_x = lambda x, y, z: y * z + 1  # x-face properties
+        tau_y = lambda x, y, z: x * z + 2  # y-face properties
+        tau_z = lambda x, y, z: 3 + x * y  # z-face properties
+
+        tau = 3 * [None]
+        for ii, comp in enumerate(["x", "y", "z"]):
+            k = np.isclose(
+                eval("self.M.faces_{}".format(comp))[:, ii], 0.5
+            )  # x, y or z location for each plane
+            tau_ii = 1e-8 * eval(
+                "np.ones(self.M.nF{})".format(comp)
+            )  # effectively zeros but stable
+            tau_ii[k] = eval("call(tau_{}, self.M.faces_{}[k, :])".format(comp, comp))
+            tau[ii] = tau_ii
+        tau = np.hstack(tau)
```

Comment:
After what you commented on today's SimPEG meeting regarding flake complaining about unused `tau_x`, `tau_y` and `tau_z` (although they are being used in the `eval` calls), I was thinking that using `eval` could be avoided in this case by:
- using `getattr` for the attributes and methods of objects, and
- define the lambda functions as values of a dict, and then iterate over the dictionary.

For example:
```suggestion
        tau_funcs = {
            "x": lambda x, y, z: y * z + 1,  # x-face properties
            "y": lambda x, y, z: x * z + 2,  # y-face properties
            "z": lambda x, y, z: 3 + x * y,  # z-face properties
        }

        tau = []
        for ii, (comp, tau_func_i) in enumerate(tau_funcs.items()):
            m = getattr(self, "M")
            k = np.isclose(
                getattr(m, f"faces_{comp}")[:, ii], 0.5
            )  # x, y or z location for each plane
            tau_ii = 1e-8 * np.ones(
                getattr(m, f"nF{comp}")
            )  # effectively zeros but stable
            tau_ii[k] = call(tau_func_i, getattr(m, f"faces_{comp}")[k, :])
            tau.append(tau_ii)
        tau = np.hstack(tau)
```

This also prevents violating PEP8 regarding assignments of `lambda` functions (https://peps.python.org/pep-0008/#programming-recommendations).

Reply:
this error has been added to flake8 exceptions